api.go

```diff
@@ -16,28 +16,27 @@ type api struct {
 	// result.
 	query string
 
+	// args is a struct that that the JSON request body is decoded into.
+	args func() interface{}
+
 	// value returns a pointer to a newly allocated Go variable able to
 	// represent the JSON object returned by the query.
 	value func() interface{}
-
-	// args is a struct that that the JSON request body is decoded into.
-	args struct {
-		ConferenceID int `json:"conference_id" db:"conference_id"`
-	}
 }
 
 // TODO: return an empty array instead of nothing if no results returned from database?
 
 // TODO: if args are required but not supplied, return a specific error message?
 
 func (a *api) serve(s *server) {
-	err := json.NewDecoder(s.r.Body).Decode(&a.args)
+	args := a.args()
+	err := json.NewDecoder(s.r.Body).Decode(&args)
```

Comment:
I don't think `&args` is necessary; I think just `args` should work fine here.

The `args` functions should return a pointer to the newly allocated value.